HIP/hip-0000-nft-standard.md

```diff
@@ -0,0 +1,795 @@
+
+It's recommended to host your file on [IPFS](https://ipfs.io/) and use a service like [Pinata](https://pinata.cloud/) to easily pin your file. Your CID should look like this: `ipfs://<hash>`.
+
+Alternatively, you can use [Arweave](https://www.arweave.org/), receiving a similar CID that looks like this: `ar://<hash>`.
+
+
+### sha256_checksum
+
+**Type:** string (SHA-256)
+
+**Optional (recommended)**
+
+**Description:** SHA-256 digest of the file pointed by the `image` property. The `sha256_checksum` property that contains a cryptographic hash of the representation of the resource the author expects to load. 
+
+For instance, an author may wish to load some image from a shared server. Specifying that the expected SHA-256 hash of https://example.com/image.jpeg is ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad means that the user agent can verify that the data it loads from that URL matches that expected hash before loading the NFT. This integrity verification significantly reduces the risk that an attacker can substitute malicious content.
+
+References: https://w3c.github.io/webappsec-subresource-integrity/
+
+
+### type
+
+**Type:** string (MIME type)
+
+**Required**
+
+**Description:** Mime formatting shall follow the following format: **type/subtype**. As a rule, mime types are all lowercase. However apps should be programmed to accept any case for robustness.
+
+A list of common mime types can be found here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+
+Note that mime types for directories are not uniformly defined. Some IPFS CIDs point to directories rather than files, so this type is useful. This standard shall define it using the format: text/directory.
+
+
+### format
+
+**Type:** string
+
+**Required**
+
+**Possible values:** 
+
+- `none`: When not following any standard (not recommneded)
+- `opensea`: When following the [OpenSea standard](https://docs.opensea.io/docs/metadata-standards)
+- `HIPXXX`: When following a standard defined by HIPs, e.g. `HIP412`
+
+**Description:** Describes the used standard to display NFT metadata. This property allows for faster parsing by NFT tooling because they don't have to figure out which standard they are dealing with. 
+
+
+### properties
+
+**Type:** object
+
+**Required**
+
+**Description:** JSON object that covers the overarching properties of the token. It holds all additional properties the artist wants to define for its NFT. Only the "external_url" property is an optional property you can include.
+
+**It's not allowed to add any non-specified root-level properties to the metadata. Any additional data should go into "properties".**
+
+
+### properties.external_url
+
+**Type:** string (URI)
+
+**Optional**
+
+**Description:** External URI pointing to an informational page about a collection or specific NFT information i.e. `www.mynft/collection/<serial>/`.
+
+
+### files
+
+**Type:** array
+
+**Required**
```

Comment:
I don't think this should be required. If the NFT preview image is enough to represent the NFT and there is no need for any higher res imagery, then we shouldn't force them to add any other files here.

Reply:
See other comment about forcing this. They can always set it to the same image if they really want to have the main image be the preview image. It's not a lot of extra metadata. I prefer forcing this.